NYTimes.com.js

```diff
@@ -90,14 +94,46 @@ function scrape(doc, url) {
 			}
 		}
 		item.url = ZU.xpathText(doc, '//link[@rel="canonical"]/@href') || url;
-		var pdfurl = ZU.xpathText(doc, '//div[@id="articleAccess"]//span[@class="downloadPDF"]/a[contains(@href, "/pdf")]/@href');
+		item.libraryCatalog = "NYTimes.com";
+		// Convert title/tags of NYT archives when in all caps: 
+		if (item.title == item.title.toUpperCase())
+				item.title = ZU.capitalizeTitle(item.title, true);
+		var allcaps = false;
+		for (i=0; i < item.tags.length; i++) {
+			if (item.tags[i] == item.tags[i].toUpperCase())
+				allcaps = true;
+		}
+		if (allcaps) {
+			for (i=0; i < item.tags.length; i++) {
+				item.tags[i] = ZU.capitalizeTitle(item.tags[i], true);
+			}
+		}
+		// PDF attachments are in subURL with key & signature
+		var pdfurl = ZU.xpathText(doc, '//div[@id="articleAccess"]//span[@class="downloadPDF"]/a[contains(@href, "/pdf")]/@href | //a[@class="button download-pdf-button"]/@href');
 		if (pdfurl) {
-			item.attachments.push({
-				url: pdfurl,
-				title: "NYTimes Archive PDF"
-			});
+			ZU.processDocuments(pdfurl, 
+				function(pdfDoc) {
+					authenticatedPDFURL = pdfDoc.getElementById('archivePDF').src;
+					if (authenticatedPDFURL) {
+						item.attachments.push({
+							title: "NYTimes Archive PDF",
+							mimeType: 'application/pdf',
+							url: authenticatedPDFURL
+						});
+					} else {
+						Z.debug("Could not find authenticated PDF URL");
+						item.complete();
+					}
+				},
+				function() {
```

Comment:
why do you need this function? Why not just put its contens as part of the if above, i.e. after line 122.